tools/ringmonitor.go

```diff
@@ -133,22 +135,48 @@ func (rm *ringMonitor) runOnce() time.Duration {
 		taskLogger := logger.With(zap.String("type", ringTask.typ), zap.Int("policy", ringTask.policy))
 		if err := ringTask.ring.Reload(); err != nil {
 			atomic.AddInt64(&errors, 1)
-			taskLogger.Error("couldn't not reload", zap.Error(err))
+			taskLogger.Error("could not reload", zap.Error(err))
 			continue
 		}
 		if ringTask.previousMD5 == "" {
-			ringTask.previousMD5, err = rm.aa.db.ringHash(ringTask.typ, ringTask.policy)
+			ringTask.previousMD5, ringTask.nextRebalance, err = rm.aa.db.ringHash(ringTask.typ, ringTask.policy)
 			if err != nil {
 				atomic.AddInt64(&errors, 1)
 				taskLogger.Error("couldn't retreive previous hash", zap.Error(err))
 				continue
 			}
 		}
-		if ringTask.ring.MD5() != ringTask.previousMD5 {
+		if ringTask.ring.MD5() == ringTask.previousMD5 {
+			if !ringTask.nextRebalance.IsZero() && time.Now().After(ringTask.nextRebalance) {
+				ringBuilder, ringBuilderFilePath, err := ring.GetRingBuilder(ringTask.typ, ringTask.policy)
+				if err != nil {
+					logger.Error("Could not find builder", zap.String("type", ringTask.typ), zap.Int("policy", ringTask.policy), zap.Error(err))
+					continue
+				}
+				var changedReplicas int
+				changedReplicas, _, _, err = ring.Rebalance(ringBuilderFilePath, false, false, true)
+				if err != nil {
+					logger.Error("Error while rebalancing", zap.String("type", ringTask.typ), zap.Int("policy", ringTask.policy), zap.String("path", ringBuilderFilePath), zap.Error(err))
+				}
+				if err := ringTask.ring.Reload(); err != nil {
+					atomic.AddInt64(&errors, 1)
+					taskLogger.Error("could not reload", zap.Error(err))
+				}
+				// So we don't get stuck rebalancing a ring by tiny amounts for forever:
```

Comment:
ghost told me to do this:

andrewd should notice if there are zeroed out devices in ring that couldn't rebalance to 0 because of min_part_hours and do some more rebalancing